atuin-client/src/tantivy.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{database::Database, history::History};
+use eyre::Result;
+use tantivy::{
+    directory::MmapDirectory,
+    doc,
+    schema::{Field, Schema, FAST, STORED, STRING, TEXT},
+    DateTime, Index, IndexWriter,
+};
+
+pub fn schema() -> (HistorySchema, Schema) {
+    let mut schema_builder = Schema::builder();
+
+    (
+        HistorySchema {
+            id: schema_builder.add_text_field("id", STRING),
+            command: schema_builder.add_text_field("command", TEXT | STORED),
+            cwd: schema_builder.add_text_field("cwd", STRING | FAST),
+            session: schema_builder.add_text_field("session", STRING | FAST),
+            hostname: schema_builder.add_text_field("hostname", STRING | FAST),
```

Comment:
we should do `| SLOW` so that in a future release we can swap it for `FAST` and claim 1000x performance boost 🚀